pkg/vm/emit/emit.go

```diff
@@ -32,20 +33,27 @@ func Bool(w *io.BinWriter, ok bool) {
 	Opcode(w, opcode.PUSHF)
 }
 
+func padRight(s int, buf []byte) []byte {
+	if len(buf) == s {
```

Comment:
`if len(buf) == s` then `buf[:s]` == `buf` and this condition doesn't make much sense. I think we can actually easily get away without `padRight`.

Reply:
It is needed for negative numbers. I have implemented this and added the test.